### PR TITLE
Add daily quiz component

### PR DESCRIPTION
**Pull request by Tayrika**
### Updates
- Added a daily quiz component to engage users with Space and NASA trivia:
  - Includes Quiz about Space and NASA's Europa Clipper mission
  - Users can answer 5 questions per day for a total of 35 questions over 7 days
  - Correct answer turns green, incorrect answer turns red
  - A progress bar tracks the user's current progress
  - After completing the quiz, the total score is displayed 
  - "Retry Quiz" button allows the user to retake the current day's quiz and "Next Quiz" button allows the user to take the next day's quiz

### QA
- The feature uses the user's system date to determine the current day. This needs testing across devices and timezones to confirm reliability. Alternative methods may need exploration if issues arise 
- Design could be refined - any feedback is welcome!

### Preview
Pasted Graphic 61
Pasted Graphic 60

### CODE REVIEW

File: src/components/DailyQuiz.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  // Unlocks quizzes based on the system's current date
+  useEffect(() => {
+    const today = new Date().toLocaleDateString(); // Format: MM/DD/YYYY
+    const lastCompletedDate = localStorage.getItem("quizLastCompletedDate");
+
+    if (lastCompletedDate !== today) {
+      const nextDay = currentDay + 1;
+      setCurrentDay(nextDay < quizData.length ? nextDay : currentDay);
+      setQuizDayIndex(nextDay - 1); // Set the active quiz to the latest unlocked day
```
Comment (jucheng925) [inline src/components/DailyQuiz.jsx:22]: should setQuizDayIndex also have the same conditional statement as currentday on the previous line?

**Review by jucheng925 — APPROVED** [review]

**Review by PsYk3s — APPROVED** [review]
Great job, works quite well!

File: src/components/DailyQuiz.jsx (hunk 1 above)
Comment (Tayrika) [reply src/components/DailyQuiz.jsx]: I fixed it. Thanks for the review!

**Comment by Tayrika** [comment]
I changed the color of the buttons and the progress bar. Here's the updated design
Pasted Graphic 69

**Pull request merged**